### ISSUE TITLE
I cant run Bot in my Local and have a question 

### ISSUE THREAD

**Issue by andikapratamaputra99** [AntenehAdmasu/Facebook-Webhook#1]
Hi @AntenehAdmasu 👋

I'm Dika, Software Engineer at AdaKerja. Thanks for the submission. It looks great 🔥  and Sorry if you waiting for our response.

I have an issue when running Facebook-Webhook on Local.

Here the error

![Screen Shot 2022-04-06 at 16 06 31](https://user-images.githubusercontent.com/67096702/161928668-963facb3-0cd6-4108-8acd-c2b14426d165.png)

My bot currenttly using Facebook api V7, what is yours so i can retest it, since it might be having different returned value.
 I see also you have Host it on Heroku but can i have the `VERIFY_TOKEN` for that webhook so i cann try it?

And last Is there any reason you not creating any `unit test` ?

thank you

**Comment by AntenehAdmasu** [comment]
Hi Dika,

Thank you for your email.

As per the documentation here
<https://developers.facebook.com/docs/messenger-platform/getting-started/webhook-setup/>,
the code must be hosted on a public HTTPS server that meets the following
requirements:

   - HTTPS support, Self-signed certificates are not supported
   - A valid SSL certificate
   - An open port that accepts GET and POST requests

and whenever a message is sent and the webhook is called a GET request is
sent with this query parameters:

curl -X GET "localhost:1337/webhook?hub.verify_token=<YOUR_VERIFY_TOKEN>&hub.challenge=CHALLENGE_ACCEPTED&hub.mode=subscribe"

and POST requests with a message body(like the one below):

curl -H "Content-Type: application/json" -X POST
"localhost:1337/webhook" -d '{"object": "page", "entry":
[{"messaging": [{"message": "TEST_MESSAGE"}]}]}'

As I can see from the error log, the mid(message id) was not found due to a
null-pointer exception. If you are to trigger the webhook from a local
request, I think the request body should look like what Facebook uses to
call the webhook. (like the one below)

[image: image.png]

I think you are good to go if you set the verify token environment variable
and use the same in your request.
I am happy to explain more(over a call if necessary) to further clarify.

As for the unit tests, reading the documentation for the setup and
deployment took me longer than I expected and I ran out of the allocated 24
hrs for the challenge. I can add now though if you insist.

Thank you very much,
Anteneh
unit tests now if you insist.

**Comment by andikapratamaputra99** [comment]
Sure, you can add the unit test for this project.
Also can you send again the pic you attached? i cant open it
I Just follow your setup from readme and not change anything from your code. I just add the .env file, Already connect it with my own messenger. So far i test its already connect to the messenger bot , but ya everytime i send message it always returning error like that. But ya let me try again today

Also about this one `https://messenger-webhook-challenge.herokuapp.com/` wanted to try it but i need `VERIFY_TOKEN` you set when deploy it to heroku.

<img width="830" alt="image" src="https://user-images.githubusercontent.com/67096702/162100195-1b3c5b45-e43b-4d05-8a2c-b22b00561763.png">


Thank you
Best regards
Andika

**Comment by AntenehAdmasu** [comment]
Hi Dika,

The VERIFY_TOKEN value I used is "adakerjachallenge". Also, make sure to
set the variables for the database connection string, page id and page
access token. Attached is the image I sent you in the previous email.

I'll work on the unit tests today and let you know as soon as I finish.

Regards,
Anteneh
